index.js

```diff
@@ -55,16 +58,20 @@ module.exports = {
    * not. Based on that, we return a tree that contains the correct version of
    * the polyfill at the `vendor/fetch.js` path.
    */
-  treeForVendor: function() {
-    if (isFastBoot()) {
-      return treeForNodeFetch();
-    } else {
-      return treeForBrowserFetch();
-    }
+  treeForVendor: function(vendorTree) {
+    var browserTree = treeForBrowserFetch();
+    browserTree = map(browserTree, (content) => `if (typeof FastBoot === 'undefined') { ${content} }`);
+    return browserTree;
+  },
+
+  //add node version of fetch.js into fastboot package.json manifest vendorFiles array
+  updateFastBootManifest: function(manifest) {
+    var fastbootFetchPath = path.resolve(__dirname + '/assets/fastboot-fetch.js');
```

Comment:
The reason you are seeing the error is because of this line. This will give you an absolute path. The fastboot manifest takes the relative path to the `dist` or `tmp` directory from it is going to serve the assets. The reason being it serves everything off a relative path that is given to it on boot.

Therefore, this asset needs to be part of the app build assets (ie in `dist`).

My recommendation would be:
1. Move the contents of `assets/fastboot-fetch.js` to `public/fastboot-fetch.js`
2. Once it is in `public` folder, `ember-cli` will automatically build it and put it in `dist/ember-fetch/fastboot-fetch.js` after the build.
3. You can update this function to just be:
```javascript
updateFastBootManifest: function(manifest) {
  manifest.vendorFiles.push('ember-fetch/fastboot-fetch.js');
  return manifest;
}
```

See the similar changes [here](https://github.com/tomdale/ember-network/pull/20/files).

Also see the migration guide [here](https://gist.github.com/kratiahuja/d22de0fb1660cf0ef58f07a6bcbf1a1c)